oauth2server/apps/tokens/decorators.py

```diff
@@ -169,6 +170,18 @@ def _validate_grant_type(request):
                 except KeyError:
                     raise PasswordRequiredException()
 
+        # reset password grant requires username parameter
+        if grant_type == 'reset_password':
+
+            stdlogger.debug("Grant_type reset_password")
+            try:
+                username = request.POST['username']
```

Comment:
Do we only need to handle post requests?

Reply:
It handles both in the nested try and except. NGL i didn't put much thought into this just copied the way it currently does this check with other grant types as this service should be getting replaced soon anyway.